lms/lmsdb/models.py

```diff
@@ -222,41 +267,41 @@ def create_solution(
             cls.submission_timestamp.name: datetime.now(),
             cls.json_data_str.name: json_data_str,
         })
+
         # update old solutions for this exercise
-        cls.update(**{
-            cls.latest_solution.name: False,
-        }).where(
+        other_solutions: Iterable[Solution] = cls.select().filter(
             cls.exercise == exercise,
             cls.solver == solver,
             cls.id != instance.id,
-        ).execute()
+        )
+        for old_solution in other_solutions:
+            old_solution.set_state(Solution.SOLUTION_STATES.OLD_SOLUTION)
         return instance
 
     @classmethod
-    def next_unchecked(cls):
+    def next_unchecked(cls) -> Optional['Solution']:
         unchecked_exercises = cls.select().where(
-            cls.is_checked == False,  # NOQA: E712
-            cls.latest_solution == True,  # NOQA: E712
+            cls.state == Solution.SOLUTION_STATES.CREATED.name,
         )
         try:
-            return unchecked_exercises.dicts().get()
+            return unchecked_exercises.get()
         except cls.DoesNotExist:
-            return {}
+            return None
```

Comment:
I do prefer returning a one type in this cases. Is there really a good reason to do this change?

Reply:
Yes, I use the object that returns from this function to change state and so on